great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py

```diff
@@ -71,8 +71,11 @@ def _get_data_reference_list_for_asset(self, asset: Optional[Asset]) -> List[str
 
         if asset is not None:
```

Comment:
This code block is repeated in multiple places and could be replaced with a function that accepts an `Optional[Asset]` and a `base_directory`. Comment reference #dupe-01

Reply:
@donaldheppner This is the V3 logic (not ZEP; all ZEP code is currently placed the "great_expectations/experimental" and "tests/experimental" directories).  One dramatic difference between V3 and ZEP is that such checks for `asset` make no sense in ZEP, while they are the essence in V3.  Specifically, in V3, `DataConnector` discovers and instantiates assets from configuration and uses the settings configured in asset as overrides.  This makes no sense in ZEP, because the assets are explicitly created, and `DataConnector` is used in service of the asset's access needs.